### PR TITLE
added retry-join-wan option to join a existing federation setup

### PR DESCRIPTION
**Pull request by jardleex**
Hi @mrlesmithjr,

seems like I got an improvement attack on my side again.

This very nice change will allow the role user to utilize the [retry-join-wan](https://www.consul.io/docs/agent/options.html#_retry_join_wan) option of Consul. I found it to be useful after I setup a new cluster after activating ACLs. We set the the default police to deny which prevented a new cluster to assemble as it was not aware how to reach the acl_datacenter to "learn" that his token was allowed to assemble itself.

So in short if a new cluster know's how to reach the existing federation this would not have happened. The option added will allow this.

Best

Jard

### CODE REVIEW

**Comment by mrlesmithjr** [comment]
Merged

**Comment by jardleex** [comment]
#42
